### DISCUSSION TITLE
Get the score of the document during the search

### DISCUSSION THREAD

**Discussion by curquiza** [meilisearch/product#379] [Feedback & Feature Proposal]
Following [this discussion](https://github.com/meilisearch/MeiliSearch/discussions/773), our users seem to need to get the score of a document when it's retrieved by the search engine.

Feel free to share your usage and your need to help us implement the feature.

**Comment by gmourier** [comment]
@Kerollmops Would it be technically possible to implement a score from the bucket sort? I imagine a kind of weight value extracted from the position of a ranking rule.

It could permit to give the score associated with each ranking rule in the scoring details + the final score for a document.

A boolean at the time of the search would allow to request this information from Meilisearch

**Comment by MihaiNueleanu** [comment]
We've identified 2 use cases for a relevance score:

#### 1. Multi-index search
As meilisearch doesn't currently support multi-index search, having a relevancy score would enable us to run parallel searches to multiple indexes and "merge" the results based on scores. Imagine you want to search movies, actors and directors, in the autocomplete field of a website. Having a score would allow combining the 3 results meaningfully, after being queried from each index.

#### 2. Personalisation of results
The context of a given user is often used to improve the relevancy of search results by applying personalization. Small example - online shop for shoes - given that a user has previously purchased only adidas shoes, then we should boost future search result scores of adidas shoes, to increase the likelihood they'll find what they're looking for. Having access to the meilisearch score would enable similar use-cases that where extra processing/personalization is applied outside of meilisearch.  


I hope it makes sense, and I hope it's something that will be considered for development. Hopefully it's not too technically challenging, although I must admit I've already tried and failed to reverse-engineer the milli code that handles sorting 😄 . All in all, we're super happy to be using meilisearch, and I know that this feature would be an amazing opportunity for us to build on top of the search engine. 

Thank you!

**Reply by gmourier** [reply]
Hi @MihaiNueleanu 👋

Thank you for your explanations. We will work to release the multi-index search in 2022.

Regarding returning a score, it is something we have in mind. It is also information that can be useful to give information about how the engine has behaved to sort the results and help adjust the relevancy.

Do you have any idea what payload would be returned to give you the scoring information about a document? 

Thanks!

**Reply by MihaiNueleanu** [reply]
Hi @gmourier,

> Do you have any idea what payload would be returned to give you the scoring information about a document?

I'm thinking of score as a numerical property on the resulting documents. The scale of the score doesn't really matter in my mind. One easy example would be if every document had a score property between 0 and 1. But it would work the same if it was between 0 and 100. As long as we know the minimum and maximum range of the scale, we could always normalize the score to whatever we need in post-calculations.

**Comment by juchom** [comment]
I'm looking for the same feature, having the score of a document would allow extra actions on the client.

Merging the best matches of multiple indexes.
If score is too low, we could do some extra actions.

**Comment by gmourier** [comment]
Hi @juchom and @MihaiNueleanu 👋

It could interest you to know that @loiclec has [explored some ideas](https://github.com/meilisearch/product/discussions/594) that can go in the direction of the needs being expressed here.

We will be super happy to have your thoughts 🙇‍♂️

Thank you!

**Comment by macraig** [comment]
Hello everyone! 👋

Quick update on our side, we've started working on a solution to return ranking score details on search.

You can follow the updates in https://github.com/meilisearch/product/discussions/594
